---
title: Error del compilador CS1017
ms.date: 07/20/2015
f1_keywords:
- CS1017
helpviewer_keywords:
- CS1017
ms.assetid: e0902e8a-b042-4711-a8a6-83456a3f88cb
ms.openlocfilehash: 165a8a0c9e67ada65dd4dab61026e22fb5f8bb07
ms.sourcegitcommit: 986f836f72ef10876878bd6217174e41464c145a
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 08/19/2019
ms.locfileid: "69609456"
---
# <a name="compiler-error-cs1017"></a>Error del compilador CS1017
No puede haber cláusulas catch después de la cláusula catch general de una instrucción try  
  
 Un bloque `catch` que no tome ningún parámetro debe ser el último de una serie de `catch` bloques. Para obtener más información sobre excepciones, vea [excepciones y control](../programming-guide/exceptions/index.md)de excepciones.  
  
## <a name="example"></a>Ejemplo  
 El ejemplo siguiente genera la advertencia CS1017:  
  
```csharp  
// CS1017.cs  
using System;  
  
namespace x  
{  
    public class b : Exception  
    {  
    }  
  
    public class a  
    {  
        public static void Main()  
        {  
            try  
            {  
            }  
  
            catch   // CS1017, must be last catch  
            {  
            }  
  
            catch(b)  
            {  
                throw;  
            }  
        }  
    }  
}  
```
